Mathlib/Probability/Moments/SubGaussian.lean

```diff
@@ -0,0 +1,386 @@
+  The actual definition uses `Kernel.HasSubgaussianMGF`: `HasCondSubgaussianMGF` is defined as
+  sub-Gaussian with respect to the conditional expectation kernel for `m` and the restriction of `μ`
+  to the sigma-algebra `m`.
+* `HasSubgaussianMGF`: a random variable `X` has a sub-Gaussian moment generating function
+  with parameter `c` with respect to a measure `μ` if for all `t : ℝ`, `exp (t * X)`
+  is `μ`-integrable and the moment generating function of `X` is bounded by `exp (c * t ^ 2 / 2)`
+  for all `t : ℝ`.
+  This is equivalent to `Kernel.HasSubgaussianMGF` with a constant kernel.
+  See `HasSubgaussianMGF_iff_kernel`.
+
+## Implementation notes
+
+### Definition of `Kernel.HasSubgaussianMGF`
+
+The definition of sub-Gaussian with respect to a kernel and a measure is the following:
+```
+structure Kernel.HasSubgaussianMGF (X : Ω → ℝ) (c : ℝ≥0)
+    (κ : Kernel Ω' Ω) (ν : Measure Ω' := by volume_tac) : Prop where
+  integrable_exp_mul : ∀ t, Integrable (fun ω ↦ exp (t * X ω)) (κ ∘ₘ ν)
+  mgf_le : ∀ᵐ ω' ∂ν, ∀ t, mgf X (κ ω') t ≤ exp (c * t ^ 2 / 2)
+```
+An interesting point is that the integrability condition is not integrability of `exp (t * X)`
+with respect to `κ ω'` for `ν`-almost all `ω'`, but integrability with respect to `κ ∘ₘ ν`.
+This is a stronger condition, as the weaker one did not allow to prove interesting results about
+the sum of two sub-Gaussian random variables.
+
+For the conditional case, that integrability condition reduces to integrability of `exp (t * X)`
+with respect to `μ`.
+
+### Definition of `HasCondSubgaussianMGF`
+
+We define `HasCondSubgaussianMGF` as a special case of `Kernel.HasSubgaussianMGF` with the
+conditional expectation kernel for `m`, `condExpKernel μ m`, and the restriction of `μ` to `m`,
+`μ.trim hm` (where `hm` states that `m` is a sub-sigma-algebra).
+Note that `condExpKernel μ m ∘ₘ μ.trim hm = μ`. The definition is equivalent to the two
+conditions
+* for all `t`, `exp (t * X)` is `μ`-integrable,
+* for `μ.trim hm`-almost all `ω`, for all `t`, the mgf with respect to the the conditional
+  distribution `condExpKernel μ m ω` is bounded by `exp (c * t ^ 2 / 2)`.
+
+For any `t`, we can write the mgf of `X` with respect to the conditional expectation kernel as
+a conditional expectation, `(μ.trim hm)`-almost surely:
+`mgf X (condExpKernel μ m ·) t =ᵐ[μ.trim hm] μ[fun ω' ↦ exp (t * X ω') | m]`.
+
+## References
+
+* [R. Vershynin, *High-dimensional probability: An introduction with applications in data
+  science*][vershynin2018high]
+
+-/
+
+open MeasureTheory Real
+
+open scoped ENNReal NNReal Topology
+
+namespace ProbabilityTheory
+
+section Kernel
+
+variable {Ω Ω' : Type*} {mΩ : MeasurableSpace Ω} {mΩ' : MeasurableSpace Ω'}
+  {ν : Measure Ω'} {κ : Kernel Ω' Ω} {X : Ω → ℝ} {c : ℝ≥0}
+
+/-! ### Sub-Gaussian with respect to a kernel and a measure -/
+
+/-- A random variable `X` has a sub-Gaussian moment generating function with parameter `c`
```

Comment:
Could you emphasize in this definition and the other ones that this implies zero expectation? This is not obvious from the name, as subgaussianity is typically more a condition at infinity.